### PR TITLE
Rebases to /vg/station13 code.

### PR DESCRIPTION
**Pull request by PJB3005**
Bringing you hypothermia, MoMMIs, BEAMS, SM cascade, xenoarch and most importantly the R-UST.

:cl:
rscadd: Rebased to /vg/station lighting code.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm
Comment (MrStonedOne) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: ?

Comment (PJB3005) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: Not sure what to do with cap is the issue.

Comment (MrStonedOne) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: cap is strength.

or more accurately, how strong from normal (10) is the center.

**Comment by GunHog** [comment]
I kind of wish this were a serious PR about porting the lighting system, because it is visually superior to ours, in my opinion.

File: code/game/gamemodes/clock_cult/clock_structures/tinkerers_cache.dm
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/tinkerers_cache.dm]: falloff was basically a cheap way to do light strength so just making this very weak light(which should be possible with this?), will be enough

**Comment by Razharas** [comment]
>All checks have failed

As expected of /vg

File: code/game/gamemodes/clock_cult/clock_structures/tinkerers_cache.dm
Comment (PJB3005) [inline code/game/gamemodes/clock_cult/clock_structures/tinkerers_cache.dm]: I was midway through this and I didn't want to ask on IRC what the second argument was to ruin the surprise so.

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm
Comment (PJB3005) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm]: Figured as such but there's no direct analog here.

**Comment by ChangelingRain** [comment]
How does this compare to our current lighting, besides the visual difference? Mainly, can you do very weak light over a wide area, and very strong light over a small area?

**Comment by PJB3005** [comment]
@Razharas all of those are warnings I put in as TODOs

@ChangelingRain: non shit backend for one thing.

There's no direct analog for cap in baylights, there's light power which is similar.

I don't actually understand the formulas of the lighting system, Mloc wrote that.

File: code/modules/lighting/lighting_source.dm
Comment (ChangelingRain) [inline code/modules/lighting/lighting_source.dm]: Replace this and the `if(!top_atom.light_sources)` bit with `LAZYADD(src, top_atom.light_sources)` which will do the same thing

File: code/modules/lighting/lighting_source.dm
Comment (ChangelingRain) [inline code/modules/lighting/lighting_source.dm]: `if(isturf(top_atom))`

**Comment by coiax** [comment]
These changes appear to be entirely lighting engine based. Perhaps you should limit it to that? For now?

**Comment by XDTM** [comment]
it just ports the lighting, coiax

**Comment by PJB3005** [comment]
Woosh

**Comment by Incoming5643** [comment]
Anything that puts horrible gradients over our sprites is terrible and I hate it unconditionally. Nearly all lighting system I've seen that aren't ours do this. I think we have the right idea now, keeping it simple and clean.

**Comment by WJohn** [comment]
I don't know I really like the smooth lighting as opposed to blockyness. Although this may make us lose our wonderful gradual fade in/out brightness which is very pretty in and of itself.

If this includes colored lighting then I'll say yes for sure.

**Comment by GunHog** [comment]
It is about time we updated to a prettier system! Darkness could be a lot more 'atmospheric', and colored lights can let us do prettier alarms and what not!

**Comment by PKPenguin321** [comment]
This is gonna need a serious changelog.

Pretty excited for this.

**Comment by kevinz000** [comment]
Dancefloors no longer making light
:(

**Comment by Incoming5643** [comment]
From a performance standpoint I essentially cannot be a ghost and not drop a large number of frames with this on. I mean thankfully I can turn it off but still.

**Comment by francinum** [comment]
Doesn't render for living mobs.

**Comment by MrStonedOne** [comment]
it didn't render because of our plane for lighting, removing that fixed it.

**Comment by Ninja137** [comment]
But does it port their atmos
Because that's the really important thing

**Comment by JJRcop** [comment]
hey aaaaaah i'm nearly ready to pr my identity system which requires a lot of stuff... just, steer cleqr of saycode eh?

**Comment by PJB3005** [comment]
> All these people that are missing the joke.

Smh.

**Comment by GunHog** [comment]
To be honest, PJB, only a coder would get this joke. No layman could read the changes and determine that this change includes ONLY the lighting system. They just see that a lot of things changed and assume as much.

**Comment by PKPenguin321** [comment]
Oh. I knew this was lighting from the first few files changed but didn't read far enough to see the other stuff was a joke. :(

**Comment by JJRcop** [comment]
Serves me right for trusting the PR title and body

**Comment by unusualcrow** [comment]
Best title shitpost of the year right here

**Comment by Fox-McCloud** [comment]
I knowwww you're trying to be silly, @PJB3005 but I'd just change the title to avoid confusing everyone. ANyone who keeps up with git, but can't read code is going to be utterly confused by this.

@WJohn 

Yes, this has colored lights.

img

**Comment by PKPenguin321** [comment]
Sexy as fuck

**Comment by Razharas** [comment]
I dont like how it looks, it makes snow loose all detail and just look like a uniform red colour

**Comment by Fox-McCloud** [comment]
...snow?

**Comment by PJB3005** [comment]
That fire in the bottom right does kind of look like snow now that you mention it.

**Comment by Razharas** [comment]
It doesnt matter if its snow or some other white stuff covering the floor, originally it had some detail drawn on it but with this light it became straight red with nothing in it

**Comment by PKPenguin321** [comment]
Well I mean it kind of makes sense, go stare at the sun and see if you can still make out the colors of the sky around it

**Comment by PJB3005** [comment]
Also I think that may be the fire overlay itself blocking the snow somewhat too.

**Comment by GunHog** [comment]
It is probably foam, not snow!

**Comment by JJRcop** [comment]
For the record I'd like to say that I can indeed read code but I didn't, just saw the title and 123 files changed and freaked out.

**Comment by Fox-McCloud** [comment]
There is noooo snowww in the pictureee.

**Comment by PJB3005** [comment]
Whoosh.

**Comment by kilkun** [comment]
go away

**Comment by PKPenguin321** [comment]
Oh no it hurts his fragile baby eyes quick deny everybody else this very nice feature

**Comment by kilkun** [comment]
"very nice"

it makes the game look like dog dookie.

**Review by Razharas — COMMENTED** [review]
Like the edge smoothing
Coloured lights look like shit though

**Comment by lzimann** [comment]
it's kinda ugly(plus colored light sucks)

**Comment by PKPenguin321** [comment]
No kilkun you look like dog dookie

**Comment by kilkun** [comment]
shut up nerd this is a _professional coding environment_ and i wont stand for this kind of slander

**Comment by unusualcrow** [comment]
The only reason TG can't have nice things is because they won't accept them

**Comment by Incoming5643** [comment]
also they perform badly and we have vaguely higher standards for such things.

**Comment by kilkun** [comment]
because the playerbase wants gamemodes, not gimmicks.

and not reskinned copy pasted crap like clock cult or poorly thought out ones like hand of god, shadowling, gang or chrono agents.

when was the last time we had an honest to god really good gamemode introduced? over 6 years. _6 years._

**Comment by Bawhoppen** [comment]
T A C O  L I G H T I N G

no seriously

**Comment by Poojawa** [comment]
To be honest, Kilkun, what would you suggest for a game mode for the furious, fast pace feel of the server that isn't already basically covered by a different game mode?

Pretty much everything is covered I feel. The modes like gang is a blend of Malf and Rev. Shadowlings being a blend of Cult and Ling. There might not be super new and unique modes but the neat af gimmicks keep things entertaining.

**Comment by kilkun** [comment]
gang has absolutely zero room for any kind of gimmicks due to the design of the mode, and its not really fun to play as a gangster in, you get and uzi and a stetchkin and thats the extent of your fun toys. extremely fucking lame. it seems like it more exists for security players to flex their dicks against laughable antags.

rev only works because it's very simple and its a very fast gamemode (except when its not), and all other conversion modes miss this fact. if you get converted, your round is completely interrupted and that fucking sucks when a round lasts more than 20 minutes.

shadowlings are shit because they are broken to hell and you have to really be shit at the game to fail as one, and also its yet another conversion mode.

consider which gamemodes work the best: nuclear emergency, traitor, changeling 1.0 sans parasting and wizard. traitor and ling 1.0 worked because they were designed to invoke paranoia and you had no idea who was out to get you until they whipped out a .357 or their proboscis. Nuke and wizard work because it involves a lot of action and firepower concentrated in a small amount of players and they generally don't drag on. Both types allowed you to go about your round without too much interruption... at least until the antagonists put you on their list of shit to fuck up.

theoretically the only things in the game right now that can work as a gamemode is abductors with minor tweaks (functioning as a far superior play on the idea of multiple teams of antagonists trying to outdo each other) and ninja with a rework (IE power consumption lowered, more tools and gadgets, return of instant gibbing) could function well as a slippery glass cannon a la wizard.

the only way to make gang work is to make them a secondary threat to something far more sinister, remove their whole dominator schtick and give them more of an incentive for fighting gang wars.

shadowling is best left as an admin event or even pitting the 3 slings against each other. though even that might not fix the fucking mode.

**Comment by Tacolizard** [comment]
@Bawhoppen are you talking about me

ALSO DID SOMEBODY SAY LIGHTING?!!!

JUST LETTIN YALL KNOW THAT A LARGE PORTION OF THE SS13 PLAYERBASE EXPERIENCES GRAPHICAL ISSUES WITH THE /VG/GOON LIGHT SYSTEM THAT MAKE THE GAME **UNPLAYABLE!!**

THE FIX IS TO INSTALL THE JUNE 2010 DIRECTX SDK, WE SHOULD BE ADVERTISING THAT FIX CURRENTLY IN-GAME BECAUSE IT FIXES PARALLAX BUGS, AND ESPECIALLY IF THIS GETS MERGED.

**Comment by Tacolizard** [comment]
also i cant be fucked to look through your changes, but did you properly make consoles give off colored light

**Comment by PJB3005** [comment]
@Tacolizard I literally PR'd this as soon as it was mostly functional.

Reason I haven't worked on it much is because I've been prioritizing UTF-8 more.

**Comment by Bawhoppen** [comment]
@Tacolizard I'm referring to a yog coder who coded/is coding a lighting system that uses actually good colored lighting

**Comment by PJB3005** [comment]
Literally if the coloured lighting is too saturated it's the issue of all devices using too deeply coloured and strong lighting colours.

**Comment by optimumtact** [comment]
stale

**Comment by GunHog** [comment]
@PJB3005 Do not let this be closed! Are you still working on this?

**Comment by PJB3005** [comment]
I'm prioritizing some other things and I can't really work on it actively due to IRL stuff so...

I won't let you down though.

**Comment by MrStonedOne** [comment]
If you don't close the pr, oranges will, and then you'll have to nag him to re-open it

**Comment by PJB3005** [comment]
Oh no the wrath of the orange

**Comment by MrStonedOne** [comment]
I'm just saying, its easier to re-open it yourself then find and get a maintainer to do so.

**Comment by optimumtact** [comment]
if he says he's busy I can accept that and will let the pr sit for a while

**Comment by unusualcrow** [comment]
RIP

**Comment by PJB3005** [comment]
dreamseeker_2017-02-04_00-49-39

Progress:tm:

**Comment by Bill-Luxe** [comment]
What the fuck?

**Comment by PJB3005** [comment]
Real screenshot:

**Comment by Saegrimr** [comment]
W A N T

**Comment by GunHog** [comment]
FINISH THIS!!!!! :D

**Comment by PKPenguin321** [comment]
oh that's so good

**Comment by GunHog** [comment]
I cannot test this while there are merge conflicts :(

**Comment by gt33** [comment]
FIX THIS AND MERGE IT!!! I MUST HAVE THAT NEW LIGHTING SYSTEM!!!!!

**Comment by gt33** [comment]
Does this add the Space Pods, @PJB3005?

**Comment by PJB3005** [comment]
Woosh.

**Comment by gt33** [comment]
XKCD reader?

File: code/game/turfs/simulated/floor/plating/lava.dm
Comment (ChangelingRain) [inline code/game/turfs/simulated/floor/plating/lava.dm]: light_range = 1 followed by light_range = 2

Comment (PJB3005) [inline code/game/turfs/simulated/floor/plating/lava.dm]: Whoops.

**Comment by GunHog** [comment]
Oh sweet, I can finally test this now!

**Comment by PJB3005** [comment]
@GunHog no you can't

**Comment by RemieRichards** [comment]
@PJB3005 is this not ready, or are we just waiting on bureaucracy and testmerges?

**Comment by PJB3005** [comment]
@RemieRichards uhm

**Comment by RemieRichards** [comment]
Awesome, ship it.

(Oh dear.)

**Comment by GunHog** [comment]
I was completely blind as ghost, human, and AI. So nope, this is sadly not ready yet :disappointed:

**Comment by PJB3005** [comment]
Alright here's the problem.

The code works and everything is fine, except for ✈️.

I need to have a constant backdrop of white behind the lighting plane for technical reasons, but for some reason, the lighting overlays don't seem to appear at all with the backdrop there. Even when on `OVERLAY`.

Any help would be seriously appreciated here, I can't get this shit to work.

File: code/modules/lighting/lighting_screen.dm
Comment (MrPerson) [inline code/modules/lighting/lighting_screen.dm]: layer = BACKGROUND_LAYER

You're welcome

Comment (GunHog) [inline code/modules/lighting/lighting_screen.dm]: I tried this locally, it just makes everything black.

Comment (PJB3005) [inline code/modules/lighting/lighting_screen.dm]: It works!

**Comment by PJB3005** [comment]
Unless you want me to give things coloured lighting SHIP IT.

File: code/modules/lighting/lighting_screen.dm
Comment (GunHog) [inline code/modules/lighting/lighting_screen.dm]: Why did it not work for me!?

**Comment by PJB3005** [comment]
Nevermind shit's broke badly I need more time to debug this.

**Comment by GunHog** [comment]
Is everything black again?

File: code/modules/lighting/lighting_source.dm
Comment (PJB3005) [inline code/modules/lighting/lighting_source.dm]: @ChangelingRain did I do this correctly because I think this is causing everything to break.

**Comment by PJB3005** [comment]
@GunHog no the lighting stuff renders correctly now but the backend logic of the light sources is broken badly.

**Comment by GunHog** [comment]
But everything is still black! Am I doing something wrongly?

**Comment by PJB3005** [comment]
Check my latest two commits.

**Comment by GunHog** [comment]
@PJB3005	git rebase --continue			0d30178
 @PJB3005	Merge remote-tracking branch 'upstream/master' into rebase-to-vg

Are the ones I have merged to my test branch. I load it up and play, everything is pitch black.

**Comment by PJB3005** [comment]
Alright ship it JUST fixed everything.

Quick once #24027 is merged test merge it!

**Comment by PJB3005** [comment]
Travis is failing because it's still using 510 to compile. It'll work if you switch to 511.

File: code/modules/lighting/lighting_screen.dm
Comment (Cyberboss) [inline code/modules/lighting/lighting_screen.dm]: No, just fix the statement

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/lighting/lighting_screen.dm
Comment (RemieRichards) [inline code/modules/lighting/lighting_screen.dm]: 511 should be the stable by now, we should just move over.

Comment (AnturK) [inline code/modules/lighting/lighting_screen.dm]: Still not stable, but we should move anyway

**Comment by GunHog** [comment]
How do I set color to things?

File: code/modules/lighting/lighting_screen.dm
Comment (MrPerson) [inline code/modules/lighting/lighting_screen.dm]: There's several fairly severe outstanding bugs

**Comment by PJB3005** [comment]
@GunHog variable is `light_color`, just edit it in VV.

File: code/modules/lighting/lighting_screen.dm
Comment (PJB3005) [inline code/modules/lighting/lighting_screen.dm]: Ok then.

**Comment by PKPenguin321** [comment]
Yesss

**Comment by XDTM** [comment]
This is so cool

**Comment by dannno** [comment]
the cut off between pitch black and the lighting is harsh

**Comment by ChangelingRain** [comment]
You can see the tiles immediately around you in the dark as well.

**Comment by PJB3005** [comment]
@ChangelingRain that's intended though, did DAL not have that?

**Comment by ChangelingRain** [comment]
It did not; it was just full black all the time.

**Comment by francinum** [comment]
Removing light sources works, creating new ones does not.

**Comment by PJB3005** [comment]
No the issue is that after the initial server init, `apply_lum` seemed to be completely incapable of finding any turfs to apply to.

Weird thing is it works perfectly locally.

**Comment by NocturnalQuill** [comment]
The shading looks pretty nice

**Comment by XDTM** [comment]
PDA light was super dim, is that intended or a bug?

**Comment by BeeSting12** [comment]
Am I the only one who liked old lighting?

**Comment by chanoc1** [comment]
yes nostalgia-nerd ;^)

**Comment by Magikarpet** [comment]
Light floor tiles dont seem to glow like computers. Is that intentional?

**Comment by PJB3005** [comment]
Probably not.

**Comment by NocturnalQuill** [comment]
It definitely needs some bugfixes, but in my opinion it is very much worth pursuing

**Comment by Kromgar** [comment]
image

Space light doesnt work? It looks rather off when you see pitch black next to the bright space tiles

**Comment by Kromgar** [comment]
image
Good fucking god i want mesons that don't make everything super bright now

**Comment by Kromgar** [comment]
Also for the lava crack tiles to emit more light because holy shit this is fucking intense

**Comment by Kromgar** [comment]
If there's a way to make the los not big black boxes this would look so much better

**Comment by PJB3005** [comment]
Big black boxes? What?

**Comment by Plapatin** [comment]
I still want MoMMIs

**Comment by RemieRichards** [comment]
@PJB3005 they're talking about the line of sight stuff built into DM, that adds pure black over what you can't see, the "void" we stick a clickcatcher over.

**Comment by ChangelingRain** [comment]
#24034
Plantpeople and shadowpeople act as though all tiles have no light at all.

**Comment by PJB3005** [comment]
Removing big black boxes is as easy as disabling soft lighting.

**Comment by PJB3005** [comment]
You know I should probably move `WHITE_THING.dmi` out of the repo root...

**Comment by Jazaen** [comment]
Please tell me this gives us VG genetics...

**Comment by GunHog** [comment]
It goes us only the lighting system. The PR name is deceptive on purpose.

**Comment by MrStonedOne** [comment]
>Travis is failing because it's still using 510 to compile. It'll work if you switch to 511.

We've been avoiding switching to 510 on travis because of the if assignment """"bug"""" (doing if (blah = blah) no longer compile errors.)

what 511 feature are you using?

**Comment by MrStonedOne** [comment]
>code/game/objects/structures/life_candle.dm:40:error: SetLuminosity: undefined proc
>code/game/objects/structures/life_candle.dm:43:error: SetLuminosity: undefined proc
it sounds more like you forgot to convert something over

**Comment by GunHog** [comment]
That is a recently merged feature, so it is a botched conflict fix.

**Comment by PJB3005** [comment]
No not even that, something got merged that touches DAL and I didn't make a fix yet so a travis build made after it fails because travis merges upstream when it does builds.

**Comment by PJB3005** [comment]
@MrStonedOne 
```dm
x = y = 10
```

File: code/modules/lighting/lighting_screen.dm
Comment (MrStonedOne) [inline code/modules/lighting/lighting_screen.dm]: The plan last i remember was to stay on 510 for Travis for as long as we can purely because of the likelihood of doing if (blah = blah) accidentally and having no indication of such a bug.

Comment (PJB3005) [inline code/modules/lighting/lighting_screen.dm]: Ah.

Thanks BYOND.

**Comment by PJB3005** [comment]
Alright, shadow and pod people are probably fixed, and soft lighting is removed.

Anything else?

**Comment by ChangelingRain** [comment]
Well, I do kind of want moving lights to not be jerky but I don't know how costly that'd be.
But besides that I think this is good!

**Comment by PJB3005** [comment]
I could give it a shot I suppose.

**Comment by GunHog** [comment]
What did you mean when you said you removed soft lighting?

**Comment by PJB3005** [comment]
Soft lighting is the "ability to barely see in dark around you effect". People didn't like it so I disabled it.

**Comment by Bawhoppen** [comment]
I'm telling you all, we should use tacolighting

**Comment by maxyo** [comment]
Hi dude. I'm ported your vg light on my fallout build which based on latest tg code.
And i solved problem with additional screen which you added. Remove it and set on master light plane multiplive blend mode. Sorry for my english. Hope you understand me

**Comment by PJB3005** [comment]
@maxyo that's the easy solution yes, but how did you solve the issue of areas that shouldn't have lighting at all being dark?

**Comment by maxyo** [comment]
@PJB3005 Sorry, i can hardly understand you. Let's speak in discord. We can test it on server.
My discord name: JackerZz#8767

**Comment by PJB3005** [comment]
Which discord server though?

**Comment by maxyo** [comment]
@PJB3005 https://discord.gg/CZgTuEq

**Comment by PJB3005** [comment]
With help from @maxyo I removed the need for another screen object.

**Comment by unusualcrow** [comment]
6 additions and 77 deletions and you fixed the issue. amazing

**Comment by maxyo** [comment]
It's working after your last commit???

**Comment by PJB3005** [comment]
@maxyo yep, perfectly.

**Comment by MrStonedOne** [comment]
fix the conflicts and we can test merge it again.

**Comment by ChangelingRain** [comment]
New space turfs don't produce space light.
`/obj/machinery/computer/mech_bay_power_console` and `/obj/machinery/computer/camera_advanced/base_construction` should have purple light
`/obj/machinery/computer/teleporter` should have blue light
`/obj/machinery/computer/slot_machine` should probably have, like, brown light
`/obj/machinery/computer/security/telescreen` should probably not have light

**Comment by KorPhaeron** [comment]
It makes lavaland very pretty so I appreciate that but I have people telling me their computers can't run the game anymore and that makes me sad

**Comment by Xhuis** [comment]
Yeah, this game being super easy to run is a big appeal. I see people posting screenshots of this game and they're using _Windows XP_. I think this is pretty but I would vote against merging it if it stopped people from playing.

**Comment by GunHog** [comment]
How many people does this affect? Is there a solution for the people running older PCs? Is there not a way that we can keep this lighting system without hurting them?

This lighting is very atmospheric and looks much better than our blocky, limited system we have no. I feel it would be no small loss to have to give this up. I want to see us continue progressing,

**Comment by KorPhaeron** [comment]
Also we need to make PDAs a bit brighter because you are more or less blind if thats all you have, can barely even see your own tile

**Comment by unusualcrow** [comment]
If your PC can't run this system then you're _seriously_ in need of an upgrade.

**Comment by PJB3005** [comment]
Didn't you guys drop IE8 support with tgui anyways?

**Comment by PJB3005** [comment]
Also, yeah. I've ran SS13 more than fine on an 11 year old laptop running Windows 8.1. I don't know what kind of shit PCs you guys are using.

**Comment by optimumtact** [comment]
We never really had IE8 support in the first place

**Comment by optimumtact** [comment]
@GunHog it's not clear but I would suspect a miniscule fraction - no more than one or two people

**Comment by optimumtact** [comment]
@Xhuis support for Windows XP ended April 8, 2014, using XP is a terrible idea, blocking people on XP so they are forced to upgrade their system is a moral requirement.

The kind of people who don't update their OS after it leaves the support window are the kind of people who are contributing to spam botnets and ddos attacks

**Comment by Xhuis** [comment]
I never said that they weren't dumb. I just said that they're part of the playerbase.

**Comment by Incoming5643** [comment]
This code runs poorly on people with old intergrated graphics card, it has very little if anything to do with the OS.

It makes me lose a healthy number of frames on Windows 7, mind I have an integrated underclocked GPU from 2009 but this game has absolutely no history of requiring an even modestly decently computer. It's at best a shitty thing to do if you don't have any way to turn it off in game.

@PJB3005 We dragged it back, kicking and screaming

**Comment by optimumtact** [comment]
your gpu is 8 years old, 8 YEARS incoming, 8 Years.

I swear to god I will buy you a new laptop and mail it to you.

I'll doit, I swear on my life I'll pull the trigger

**Comment by GunHog** [comment]
Please do Oranges, so he can stop suffering

**Comment by RemieRichards** [comment]
>Please do Oranges, so he can stop making us suffer

**Comment by Incoming5643** [comment]
I've been very hands off on this one. 

I know I don't have the strongest argument, but it's still a valid one.

**Comment by juanototo2** [comment]
Where's the supermatter cascade? I'm eagerly waiting for that to happen **every single round**

after engineers discover its antag-removing properties.

**Comment by Poojawa** [comment]
While it would be good to have some form of 'potato' quality, a minimal upgrade to your system isn't outlandish.

**Comment by Incoming5643** [comment]
It kind of is, video games don't traditionally change their minimum requirements midstream. They're an assured base line that says "yes you won't get locked out of this game in six months because someone over reached on a graphical effect".

**Comment by optimumtact** [comment]
This isn't a traditional video game, it's an MMORPG delivered online with a rolling release schedule

there's nothing traditional about it

**Comment by PJB3005** [comment]
And you're also not paying money for it so you have exactly 0 rights to anything.

**Comment by Incoming5643** [comment]
I'm not arguing that it's something we should abide by because "we're a real video game, guys!"
I'm arguing that it's something we should abide by because the industry independently figured out it was a good rule to stand by because to do otherwise would needlessly ostracize parts of their customer base.

Again I'm not even straight up saying no to the idea here, I am highlight the importance of keeping client performance requirements reasonable.

**Comment by PKPenguin321** [comment]
I disagree, games from eight years ago either weren't updated at all, are updated extremely infrequently and only for general stability (half life 1 on steam was updated a few months ago), or do require a better computer to play now (TF2)

**Comment by MrStonedOne** [comment]
The falloff is WAY too strong on most stuff

its basically full bright for x tiles then a 1 or 1.5 tile fall off.

**Comment by MrStonedOne** [comment]
like, have you ever looked at what /tg/lighting looks like in low light.

secret panel -> break all lights

compare yours with /tg/'s current

you will see that /tg/'s current is objectively better.

**Comment by MrStonedOne** [comment]
Here are the prs where i tweaked our falloff
#12748
#13333

**Comment by ChangelingRain** [comment]
>objectively better

hey look it's a mso platitude where his opinion is The Only Correct Opinion No Fuck Off I'm Right Stop Disagreeing With Me

**Comment by KorPhaeron** [comment]
It actually hurts my eyes and makes me feel vaguely uncomfortable looking at the screen like it is now, like my vision is being eaten away and I'm actually going blind around the edges of my eyes.

I'm probably crazy and I'll probably get over it.

**Comment by MrStonedOne** [comment]
>words

Are you still being a little shit?

I was hoping that if i just ignored it you'd stop, but you've had it out for me since #devbus

**Comment by PJB3005** [comment]
@MrStonedOne yeah I need to tweak the light power and range values more.

**Comment by Kromgar** [comment]
image


I dunno this has been the main test merge and bagil keeps crashing where no one can join. 

One time it happened while singulo was loose. I feel like its lighting related but i cant be sure

**Comment by ChangelingRain** [comment]
This fixes #24167

**Comment by AnturK** [comment]
Something weird going on with the light when space is exposed (or due to foam?)
pic

**Comment by RemieRichards** [comment]
Oh god that's creepy as shit

**Comment by ChangelingRain** [comment]
Nobody commented about the memory leak this has that sometimes fucks the server for, uh, some reason?
And maybe it's a good idea to alert the person making the pr that there's a problem.

**Comment by optimumtact** [comment]
I thought that was a byond bug that was reported and fixed

**Comment by Jordie0608** [comment]
http://www.byond.com/forum/?post=2213856?

**Comment by MrStonedOne** [comment]
oj, no, it happens on 1374.

**Comment by optimumtact** [comment]
Stale

**Comment by GunHog** [comment]
@PJB3005 ### DO NOT ALLOW THIS PR TO BE CLOSED GET THE CONFLICTS MERGED AND THE MEMORY LEAK FIXED PLEASE! THIS IS THE BEST UPDATE OF THE YEAR PLEASE DO NOT LET THE FRUIT KILL IT!

**Comment by PJB3005** [comment]
I'm not even in Belgium right now unless you want me to fix merge conflicts on my phone with roaming charges, rip.

**Comment by MrStonedOne** [comment]
then close it yourself so you can open it yourself

**Comment by PJB3005** [comment]
Genius!

**Comment by GunHog** [comment]
VG lights when?

**Comment by PJB3005** [comment]
When I get home tomorrow.

That said it occurred to me I could totally fix conflicts over SSH but fuck using vim on a phone.

**Comment by GunHog** [comment]
### Welcome Back! :heart:

**Comment by optimumtact** [comment]
I'll merge it man I aint playing around

**Comment by GunHog** [comment]
You have another conflict! Poor guy

**Comment by ChangelingRain** [comment]
Is the memory leak fixed yet because if it is I'm super gonna merge this

**Comment by Xhuis** [comment]
So what does this PR add exactly? People say lighting but the PR body says otherwise.

**Comment by PJB3005** [comment]
Woosh!

**Comment by XDTM** [comment]
Trust people

**Comment by Xhuis** [comment]
I assume that I'm not getting a cohesive answers then

**Comment by XDTM** [comment]
Yes, it is lighting only, which you could tell if you glanced at the code

**Comment by Xhuis** [comment]
I saw the lighting stuff but I didn't know if that was all. I don't know VG code. Thank you for the answer.

**Comment by Incoming5643** [comment]
Please make the changelog entry actually correct. It's not the place for sarcasm.

**Comment by GunHog** [comment]
>Yes, it is lighting only, which you could tell if you glanced at the code

Because we totally expect everyone on Github, including non-coders, to read through ~1300 new lines of code across 159 files to determine that the only change is lighting. 

This is akin to a renowned physicist hosting a high-school science class while having no experience as a teacher. His expectations are going to be severely off the mark.

**Comment by Poojawa** [comment]
I mean, you have to scroll completely through all comments to make a comment asking what's going on.

I'm sure he'll change it before it's merged.

**Comment by Xhuis** [comment]
[If you insist
](https://puu.sh/uu4FV/a548d9d06f.gif)

**Comment by Bawhoppen** [comment]
that is a nice hearty sized gif 

also please don't merge this

**Comment by PJB3005** [comment]
RIOT

**Comment by ChangelingRain** [comment]
You're gonna have to make it serious at some point!
And also seriously is the memory leak fixed or what

**Comment by PJB3005** [comment]
There's no memory leak from what I can tell.

**Comment by ChangelingRain** [comment]
In that case, it's time forrrrrrr

**Comment by ChangelingRain** [comment]
I sure hope you're right!

**Comment by PJB3005** [comment]
oh boy.

**Comment by GunHog** [comment]
YEEEEEEEEEEEEEEEEEEES

**Comment by Xhuis** [comment]
PogChamp

**Comment by chanoc1** [comment]
we did it boys

**Comment by KorPhaeron** [comment]
Shouldn't we have tested this more if we thought there was a memory leak

**Comment by Incoming5643** [comment]
@KorPhaeron 
Yep

**Comment by PKPenguin321** [comment]
Wooooooooooooooooo

**Comment by Bawhoppen** [comment]
worst pr of 2017 just got merged its all over year of trump

**Comment by Okand37** [comment]
Very pretty.

**Comment by PKPenguin321** [comment]
Zombie!

**Comment by GunHog** [comment]
@PJB3005 How do I disable permanent lighting in a shuttle area?

**Comment by PJB3005** [comment]
@GunHog give them `dynamic_lighting = TRUE`

**Comment by GunHog** [comment]
I did for `/area/shuttle/auxillary_base` but it seems that something at runtime is setting it back to `FALSE`.

**Comment by MrStonedOne** [comment]
just do a find for shit that sets that, i think something in shuttle code fucks with that

**Pull request merged**
